torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -825,23 +830,20 @@ def _test_group_override_backend(self, initializer):
             dist.broadcast(tensor, src=group[0], group=group_id)
             self.assertEqual(_build_tensor(2, value=0), tensor.to("cpu"))
 
-        @require_backend({"gloo", "nccl"})
-        @require_backends_available({"gloo", "nccl"})
+        @require_backend(DistTestCases.backend_feature["gpu"])
+        @require_backends_available(DistTestCases.backend_feature["gpu"])
```

Comment:
This is prior to your change. Do you know why do we need both `require_backend` and `require_backends_available` here?

Reply:
I don't think we need both. `require_backends_available` is a stricter test and checks to make sure all of the backends in the list are available, which can fail in the case where NCCL isn't available in particular. My OSS setup can't run on my CPU only system because of NCCL not showing up as available, as an example.

There are cases where it would be useful to have a check for backend availability. It doesn't seem to actually be using it as I would expect, though.

https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/distributed/distributed_test.py#L1200

I can submit a separate patch to remove some of these and clean up where both might actually be needed. The question is if we want to check for availability or just to match on the list. I'd say to match on the list AKA `require_backend`.